Tasks/JenkinsQueueJob/jobqueue.ts

```diff
@@ -0,0 +1,260 @@
+/// <reference path="../../definitions/node.d.ts"/>
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+/// <reference path="../../definitions/shelljs.d.ts"/>
+
+import tl = require('vsts-task-lib/task');
+import fs = require('fs');
+import path = require('path');
+import shell = require('shelljs');
+
+// node js modules
+var request = require('request');
+
+import task = require('./jenkinsqueuejobtask');
+import TaskOptions = task.TaskOptions;
+
+import job = require('./job');
+import Job = job.Job;
+import JobState = job.JobState;
+
+import jobsearch = require('./jobsearch');
+import JobSearch = jobsearch.JobSearch;
+
+export class JobQueue {
+    taskOptions: TaskOptions;
+
+    rootJob: Job;
+    allJobs: Job[] = [];
+    searches: JobSearch[] = [];
+
+    constructor(taskOptions: TaskOptions) {
+        this.taskOptions = taskOptions;
+    }
+
+    intervalId: NodeJS.Timer;
+    intervalMillis: number = 10;
```

Comment:
bumped it up to 100ms.  This is a tight loop that check to see if there is work to be done, rather than a poll interval.  So, from the time work is scheduled, now it will take 100 ms for the work to start.